runtime/lua/vim/iter.lua

```diff
@@ -0,0 +1,770 @@
+--- Iterator implementation.
+
+---@class Iter
+---@field next function Return the next value in the iterator
+---@field _table ?table Underlying table data (table iterators only)
```

Comment:
Just FYI in case you're not aware,  lua-ls recently added support for modifiers like `private`: `@field private _table? table`

Is the ? as prefix on the type intentional?  So far we afaik use it as property name suffix or add `|nil`

Reply:
Thanks, didn’t know about the private field modifiers.

The ? is intentional to indicate optional, although in this case it’s not needed anyway, as those fields should move to the ListIter docstring.

Reply:
> The ? is intentional to indicate optional, although in this case it’s not needed anyway, as those fields should move to the ListIter docstring.

I meant if the syntax is intentional. `_table ?table` vs `_table? table`. I think so far we use the latter but I'm not sure anymore.

Reply:
Usually `?` appears after the name or the type, not before.

So either:
```
name? type
name type?
```

Reply:
Or explicit `table|nil`; can't go wrong with that.